cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java

```diff
@@ -0,0 +1,476 @@
+                    rc = readUnicode();
+                    break;
+                case 't':
+                    rc = "\t";
+                    break;
+                case 'f':
+                    rc = "\f";
+                    break;
+                case 'r':
+                    rc = "\r";
+                    break;
+                case 'n':
+                    rc = "\n";
+                    break;
+                default:
+                    rc = Character.toString((char) ec);
+                    break;
+            }
+            peekChar = bufferedReader.read();
+            return rc;
+        }
+
+        private String readUnicode() throws IOException {
+            int sum = 0;
+            for (int h = 0; h < 4; ++h) {
+                int hexIt;
+                peekChar = bufferedReader.read();
+                if (peekChar >= '0' && peekChar <= '9') {
+                    hexIt = peekChar - '0';
+                } else if (peekChar >= 'a' && peekChar <= 'f') {
+                    hexIt = peekChar - 'a' + 10;
+                } else if (peekChar >= 'A' && peekChar <= 'F') {
+                    hexIt = peekChar - 'A' + 10;
+                } else {
+                    throw new IllegalArgumentException("Malformed \\uxxxx encoding");
+                }
+                sum = sum * 16 + hexIt;
+            }
+            return Character.toString((char) sum);
+        }
+
+        /*
+         * On entry, peekChar is at comment char
+         * On exit, peekChar is at newline or EOF
+         */
+        private void ignoreToEndOfLine() throws IOException {
+            for (; ; ) {
+                peekChar = bufferedReader.read();
+                if (isEol()) {
+                    break;
+                }
+            }
+        }
+
+        /*
+         * On entry, peekChar is at NO_CHAR or after key
+         * On exit, peekChar is at non-whitespace, newline or EOF
+         */
+        private void ignoreWhitespace(boolean includeNewLine) throws IOException {
+            for (; ; peekChar = bufferedReader.read()) {
+                switch (peekChar) {
+                    case '\r':
+                    case '\n':
+                        if (!includeNewLine) {
+                            return;
+                        }
+                    case '\t':
+                    case '\f':
+                    case ' ':
+                    case NO_CHAR:
+                        continue;
+                    default:
+                        return;
+                }
+            }
+        }
+
+        private boolean isComment() {
+            switch (peekChar) {
+                case '!':
+                case '#':
+                    return true;
+            }
+            return false;
+        }
+
+        private boolean isKeySeparator() {
+            switch (peekChar) {
+                case '=':
+                case ':':
+                    return true;
+            }
+            return false;
```

Comment:
> The key contains all of the characters in the line starting
> with the first non-white space character and up to, but not
> including, the first unescaped {@code '='},
> {@code ':'}, or white space character other than a line
> terminator. 

I might be wrong, but I don't think you're dealing with the white space case.  Could you at least add it to the tests?

Reply:
I have added additional unit test case.